### PR TITLE
Update reference to cookies in nextjs examples

### PR DESCRIPTION
**Pull request by silentworks**
## What kind of change does this PR introduce?

Docs update for Next.js cookie setup

## What is the current behavior?

The build process for Next.js fails when it cannot detect that a cookie is being used in a server route.

## What is the new behavior?

The build process for Next.js is successful because the cookies function is in the scope of the bundler, so it knows that server route is dynamic.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx [hunk 1]
```diff
@@ -171,7 +171,8 @@ export async function GET(request) {
   const code = requestUrl.searchParams.get('code')
 
   if (code) {
-    const supabase = createRouteHandlerClient({ cookies })
+    const cookieStore = () => cookies()
+    const supabase = createRouteHandlerClient({ cookies: () => cookieStore })
```
Comment (gregnr) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx:175]: Nice. Confirming this should be a double callback?

ie:
```ts
createRouteHandlerClient({ cookies: () => () => cookies() })
```

Comment (silentworks) [reply apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx:175]: Good catch, that is indeed incorrect. This is what find and replace does after replacing one before running it.

Comment (silentworks) [reply apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx:175]: Oh wait, they are all wrong.

**Review by gregnr — APPROVED** [review]

**Comment by gregnr** [comment]
@silentworks should the `with-supabase` Next.js example template be updated?

https://github.com/vercel/next.js/blob/3c326ea5e822bd97aaa08313bbda246124a37fd3/examples/with-supabase/app/page.tsx#L43

**Pull request merged**